framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileElementTypes.java

```diff
@@ -628,15 +627,23 @@ private void parseEnclosingClass(Element e) {
             return;
         }
         String className = getOutermostEnclosingClass(e);
-        if (className == null || className.isEmpty()) {
+        if (className == null || className.isEmpty() || processingClasses.contains(className)) {
             return;
         }
-        if (jdkStubFiles.containsKey(className)) {
-            parseJdkStubFile(jdkStubFiles.get(className));
-            jdkStubFiles.remove(className);
-        } else if (jdkStubFilesJar.containsKey(className)) {
-            parseJdkJarEntry(jdkStubFilesJar.get(className));
-            jdkStubFilesJar.remove(className);
+
+        boolean prevParsing = parsing;
+        processingClasses.add(className);
+        try {
+            if (jdkStubFiles.containsKey(className)) {
+                parseJdkStubFile(jdkStubFiles.get(className));
+                jdkStubFiles.remove(className);
```

Comment:
Unrelated to your change: I think you can combine the `get/remove` calls in just calling `remove` as argument to `parseJdkStubFile`. Same below.

Reply:
There are other places where `parseJdkStubFile` is called, so maybe we should leave it as it is for simplicity.